src/search/search_algorithms/eager_search.cc

```diff
@@ -277,10 +282,11 @@ SearchStatus EagerSearch::step() {
                 */
                 open_list->insert(succ_eval_context, succ_state.get_id());
             } else {
-                // If we do not reopen closed nodes, we just update the parent pointers.
-                // Note that this could cause an incompatibility between
-                // the g-value and the actual path that is traced back.
-                succ_node.update_parent(*node, op, get_adjusted_cost(op));
+                // If we do not reopen closed nodes, we just update the parent
+                // pointers. Note that this could cause an incompatibility
+                // between the g-value and the actual path that is traced back.
+                succ_node.update_open_node_parent(*node, op,
```

Comment:
Isn't the node closed at this point? This would make the assertion inside `update_open_node_parent` fail.